class/practice_4.1/pizza.cpp

```diff
@@ -0,0 +1,68 @@
+#include "pizza.hpp"
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <map>
+
+Pizza::Pizza() : total_price(0.0) {}
+
+void Pizza::addIngredient(const Ingredient& ingredient) {
+    ingredients.push_back(ingredient);
+    total_price += ingredient.price;
+}
+
+double Pizza::getTotalPrice() const {
+    return total_price;
+}
+
+const std::vector<Ingredient>& Pizza::getIngredients() const {
+    return ingredients;
+}
+
+PizzaBuilder::PizzaBuilder(const std::string& ingredientsFile) : ingredientsFile(ingredientsFile) {}
+
+Pizza PizzaBuilder::buildPizza() {
+    Pizza pizza;
+    loadIngredients();
+
+    std::cout << "Available ingredients:\n";
+    for (const auto& pair : ingredients) {
+        std::cout << "- " << pair.second.name << " ($" << pair.second.price << ")\n";
+    }
+
+    std::string input;
+    while (true) {
+        std::cout << "Enter ingredient name (or 'done' to finish): ";
+        std::getline(std::cin, input);
+        if (input == "done") {
+            break;
+        }
+
+        auto it = ingredients.find(input);
+        if (it != ingredients.end()) {
+            pizza.addIngredient(it->second);
+        } else {
+            std::cout << "Invalid ingredient name. Please choose from the list.\n";
+        }
+    }
+
+    return pizza;
```

Comment:
as per comment above pizza should be a member of builder class with api to build different ingredients. therefore getPizza should return built pizza and reset member(local variable and std::swap can be used for that purpose)